src/serialization/interval_log/mod.rs

```diff
@@ -0,0 +1,588 @@
+
+    fn write_comment(&mut self, s: &str) -> io::Result<()> {
+        write!(self.writer, "#{}\n", s)
+    }
+
+    fn write_histogram<T: Counter>(
+        &mut self,
+        h: &Histogram<T>,
+        start_timestamp: f64,
+        duration: f64,
+        tag: Option<Tag>,
+        max_value_divisor: f64,
+    ) -> Result<(), IntervalLogWriterError<S::SerializeError>> {
+        self.serialize_buf.clear();
+        self.text_buf.clear();
+
+        if let Some(Tag(s)) = tag {
+            write!(self.text_buf, "Tag={},", &s).expect("Writes to a String can't fail");
+        }
+
+        write!(
+            self.writer,
+            "{}{:.3},{:.3},{:.3},",
+            self.text_buf,
+            start_timestamp,
+            duration,
+            h.max() as f64 / max_value_divisor // because the Java impl does it this way
+        )?;
+
+        self.text_buf.clear();
+        let _len = self.serializer
+            .serialize(h, &mut self.serialize_buf)
+            .map_err(|e| IntervalLogWriterError::SerializeError(e))?;
+        base64::encode_config_buf(&self.serialize_buf, base64::STANDARD, &mut self.text_buf);
+
+        self.writer.write_all(self.text_buf.as_bytes())?;
+        self.writer.write_all(b"\n")?;
+
+        Ok(())
+    }
+}
+
+/// A tag for an interval histogram.
+///
+/// Tags are just `str`s that do not contain a few disallowed characters: ',', '\r', '\n', and ' '.
+///
+/// To get the wrapped `str` back out, use `as_str()` or the `Deref<str>` implementation
+/// (`&some_tag`).
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub struct Tag<'a>(pub &'a str);
+
+impl<'a> Tag<'a> {
+    /// Create a new Tag.
+    ///
+    /// If a disallowed character is present, this will return `None`.
+    pub fn new(s: &'a str) -> Option<Tag<'a>> {
+        if s.chars()
+            .any(|c| c == ',' || c == '\r' || c == '\n' || c == ' ')
+        {
+            None
+        } else {
+            Some(Tag(s))
+        }
+    }
+
+    /// Returns the tag contents as a str.
+    pub fn as_str(&self) -> &'a str {
+        self.0
+    }
+}
+
+impl<'a> ops::Deref for Tag<'a> {
+    type Target = str;
+
+    fn deref(&self) -> &Self::Target {
+        self.as_str()
+    }
+}
+
+/// An individual interval histogram parsed from an interval log.
+#[derive(PartialEq, Debug)]
+pub struct IntervalLogHistogram<'a> {
+    tag: Option<Tag<'a>>,
+    start_timestamp: f64,
+    duration: f64,
+    max: f64,
+    encoded_histogram: &'a str,
+}
+
+impl<'a> IntervalLogHistogram<'a> {
+    /// Tag, if any is present.
+    pub fn tag(&self) -> Option<Tag<'a>> {
+        self.tag
+    }
+
+    /// Timestamp of the start of the interval in seconds.
+    ///
+    /// The timestamp may be absolute vs the epoch, or there may be a `StartTime` or `BaseTime` for
+    /// the log, in which case you may wish to consider this number as a delta vs those timestamps.
+    /// See the module-level documentation about timestamps.
+    pub fn start_timestamp(&self) -> f64 {
```

Comment:
Could add a method to return a `chrono` `NaiveDateTime`, but it didn't seem worth the dependency to do something that the consumer can just as easily do

Reply:
Also, given that it might be entirely user-defined, I'm not sure we should.